apps/edge-api/cmd/server/main.go

```
@@ -628,15 +628,34 @@ const metricsListenAddr = ":9102"
 }
 
-func handleHealth(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
-	json.NewEncoder(w).Encode(map[string]string{"status": "ok"})
+func handleHealth(healthy func() bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		if healthy != nil && !healthy() {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			json.NewEncoder(w).Encode(map[string]string{
```

golangci-lint (2.12.2)

[error] Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)